curator-framework/src/test/java/org/apache/curator/framework/imps/TestReconfiguration.java

```diff
@@ -313,6 +313,38 @@ public void testNewMembers() throws Exception
         }
     }
 
+    @Test
+    public void testConfigToConnectionStringNormal() throws Exception
+    {
+        String config = "server.1=10.1.2.3:2888:3888:participant;10.2.3.4:2181";
+        String configString = EnsembleTracker.configToConnectionString(toQuorumVerifier(config.getBytes()));
+        Assert.assertEquals("10.2.3.4:2181", configString);
+    }
+
+    @Test
+    public void testConfigToConnectionStringNoClientAddr() throws Exception
+    {
+        String config = "server.1=10.1.2.3:2888:3888:participant;2181";
+        String configString = EnsembleTracker.configToConnectionString(toQuorumVerifier(config.getBytes()));
+        Assert.assertEquals("10.1.2.3:2181", configString);
+    }
+
+    @Test
+    public void testConfigToConnectionStringWildcardClientAddr() throws Exception
+    {
+        String config = "server.1=10.1.2.3:2888:3888:participant;0.0.0.0:2181";
```

Comment:
It doesn't look like Zookeeper supports IPv6 addresses in 3.5.1-alpha (which is what Curator seems to be using). This appears to be fixed in Zookeeper 3.5.2 and 3.6.0:

https://issues.apache.org/jira/browse/ZOOKEEPER-1460